arch/xtensa/core/fatal.c

```diff
@@ -78,49 +79,15 @@ static char *cause_str(unsigned int cause_code)
 	default:
 		return "unknown/reserved";
 	}
-}
-#endif
-
-static inline unsigned int get_bits(int offset, int num_bits, unsigned int val)
-{
-	int mask;
-
-	mask = BIT(num_bits) - 1;
-	val = val >> offset;
-	return val & mask;
-}
-
-static void dump_exc_state(void)
-{
-#if defined(CONFIG_PRINTK) || defined (CONFIG_LOG)
-	unsigned int cause, ps;
-
-	cause = get_sreg(EXCCAUSE);
-	ps = get_sreg(PS);
-
-	z_fatal_print("Exception cause %d (%s):", cause, cause_str(cause));
-	z_fatal_print("  EPC1     : 0x%08x EXCSAVE1 : 0x%08x EXCVADDR : 0x%08x",
-		      get_sreg(EPC_1), get_sreg(EXCSAVE_1), get_sreg(EXCVADDR));
-
-	z_fatal_print("Program state (PS):");
-	z_fatal_print("  INTLEVEL : %02d EXCM    : %d UM  : %d RING : %d WOE : %d",
-		      get_bits(0, 4, ps), get_bits(4, 1, ps),
-		      get_bits(5, 1, ps), get_bits(6, 2, ps),
-		      get_bits(18, 1, ps));
-#ifndef __XTENSA_CALL0_ABI__
-	z_fatal_print("  OWB      : %02d CALLINC : %d",
-		      get_bits(8, 4, ps), get_bits(16, 2, ps));
+#else
+	return "na";
 #endif
```

Comment:
shouldn't just return null if we don't have printk nor log available ?

Reply:
Then I'd have to add extra code to optionally log this via z_fatal_print() (which in theory can exist even when printk/log don't) this instead of just printing the string it returns.  Seems like this is smaller and simpler.